lib/sycamore/sycamore/connectors/elasticsearch/elasticsearch_writer.py

```diff
@@ -120,19 +124,29 @@ def from_doc(cls, document: Document, target_params: BaseDBWriter.TargetParams)
             "bbox": document.bbox.coordinates if document.bbox else None,
             "shingles": document.shingles,
         }
-        droperties = drop_types(properties, drop_empty_lists=True, drop_empty_dicts=True)
-        if target_params.flatten_properties:
-            # Property names must be [a-zA-Z][_0-9a-zA-Z]{0,230}, so use __ as a separator rather than .
-            droperties = dict(flatten_data(droperties, allowed_list_types=[int, str, bool, float], separator="__"))
-            droperties = {k.replace("-", "_"): v for k, v in droperties.items()}  # e.g. properties__y-index is invalid
-        assert isinstance(droperties, dict)
-        return ElasticDocumentRecord(doc_id=doc_id, properties=droperties, embeddings=embedding)
+        return ElasticDocumentRecord(doc_id=doc_id, properties=properties, embeddings=embedding)
 
 
 def _narrow_list_of_doc_records(records: list[BaseDBWriter.Record]) -> TypeGuard[list[ElasticDocumentRecord]]:
     return all(isinstance(r, ElasticDocumentRecord) for r in records)
 
 
+def compare_nested_dicts(dict1: dict[Any, Any], dict2: dict[Any, Any]):
+    for key, value in dict1.items():
+        if value == "object":
+            continue
+        if key not in dict2:
+            return False
+        if isinstance(value, dict):
+            if not isinstance(dict2[key], dict):
+                return False
+            if not compare_nested_dicts(value, dict2[key]):
+                return False
+        elif dict1[key] != dict2[key]:
+            return False
+    return True
```

Comment:
prob move to common.
also in opensearch I ran into an issue where I could put in a key like so
```json
{
  "index.knn": "true"
}
```
and it would come out as 
```json
{
  "index": {
    "knn": "true"
  }
}
```
You can check out the opensearch compat function for the shenanigans I pulled to make it work, but the point is, I'm worried about elastic having similar issues

Reply:
You're right good catch, I hadn't tested it on this. Have updated it to use the opensearch code